### PR TITLE
Improved PBF Ingest Documentation

### PR DESCRIPTION
**Pull request by MikeGost**
### Description:

~~This is a DNM because I need to tweak some of the aesthetics in the README file (add links and highlight class names). However, I wanted an early review of the actual content.~~

1. The main change - added additional documentation in the form of README files. There is one for the overall PBF ingest and raw Atlas creation flow, one for the slicing logic and another for the sectioning.
2. Renamed the temporary entity package from `org.openstreetmap.atlas.geography.atlas.raw.slicing.temporary` to `org.openstreetmap.atlas.geography.atlas.raw.temporary`. This seemed more appropriate since temporary entities are not specific to slicing, but are part of a common pattern in both slicing and sectioning code.
3. Updated the `RawAtlasIntegrationTest` to make use of the single country constructor instead of the many country one. This makes it easier to read and follow.

### Potential Impact:

None - the only potential impact is the package rename, but no other code depends on it other than the classes updated in this PR.

### Unit Test Approach:

All tests pass before and after. Most changes are README file additions.

### Test Results:

N/A
------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/geography/atlas/raw/creation/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# `Raw Atlas Creation`
+
+## Overview
+
+This package is responsible for the initial OSM PBF file to Atlas file transformation. The result is an intermediate atlas file, that is not country-sliced or way-sectioned. The advantage to this is that we get to leverage the Atlas API layer and most importantly, the spatial index to make slicing and sectioning as straight-forward as possible.
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/raw/creation/README.md]: Nit: probably should be capitalized Atlas to be consistent

File: src/main/java/org/openstreetmap/atlas/geography/atlas/raw/creation/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+There are a couple of implementation details to call out for the raw Atlas creation. The input protobuf file is created using the Osmosis library and it's structured with a distinct order - the file contains the Nodes first, then the Ways and lastly the Relations. Each Way references the Node identifiers that are used to construct itself. This is something problematic, because we have no Node location or tag properties of the indiviual Nodes when processing each Way. To solve this, we must either create a Node map or make two passes over the PBF file - once to read the ways and a second time to read the Nodes for the Ways we're interested in. It's a lot faster to read the file twice, rather than resize the underlying PackedAtlas arrays during build time. In the actual implementation, the OsmPbfCounter class is responsible for identifying what to bring in, keeping track of relevant Nodes and counts. The OsmPbfReader will then go through the file a second time and build the raw Atlas using the information from the counter.
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/raw/creation/README.md]: Nit: "indiviual Nodes" should be "individual Nodes"

File: src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+2. Stream all the input Lines and identify the ones that becomes Edges or Areas. If a Line is set to become an Edge, we identify all location-based Nodes for that Edge. A Node will be created in one of the 4 cases:
+	a. A self-intersection - there is a repeated non-consecutive intersection in the Edge.
+	b. Configuration-based sectioning dependent on tagging (ex: section at barriers). 
+	c. Intersection with another edge of the same layer
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/README.md]: Nit: edge -> Edge

File: src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	d. Intersection with another edge of a different layer that either starts or ends at point of intersection
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/README.md]: Nit: edge -> Edge

File: src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/README.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+	
+3. Stream all the input Points and distinguish between the ones that are shape points of underlying Atlas entities versus ones that become separate Atlas Points in the final Atlas. Most Points will become shape points and will be tracked by the underlying PolyLine or Polygon representation of the Area, Line or Edge that it's a part of.
+
+4. Stream all the Lines that have been identified as Edges from step 2 and split them into one or more Edge. This step will take care of both open and closed edges and handle forward and reverse Edge creation. It's important to note that the logic for sectioning closed Edges is slightly more complex since we try to avoid sectioning at the arbitrary start location where the OSM Way was drawn from. Instead, we try to section at any intersecting location with other edges or if those don't exist, we have a single Edge that starts and stops at the first Location of the underlying PolyLine.
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/README.md]: Nit: "open and closed edges", "intersecting location with other edges" -> "open and closed Edges", "intersecting location with other Edges"

File: src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/README.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+For starters, we define an open OSM Way as a linear feature which does not share a first and last node. Conversely, a closed OSM Way has the last node shared with its first node. In general, the OSM Way to Atlas Entity translation is largely configuration driven. Please see org/openstreetmap/atlas/geography/atlas/pbf resource package for a full view of default configurations. However, the generally accepted convention supported by the rest of the code base is that any navigable, open OSM Way (ex: highway, ferry, pier) becomes an Atlas Edge. An OSM Way that's non-navigable and open (ex: power line, wall) becomes an Atlas Line and anything that's non-navigable and closed becomes an Atlas Area.
+
+### OSM Relation to Atlas Relation
+
+An OSM Relation maps directly to an Atlas Relation. An Atlas Relation can be any combination of Atlas Points, Nodes, Edges, Lines and Areas.
+
+## Code Sample
+
+The sectioning code is pretty simple to use. It supports two basic use cases:
+1. Given a sliced raw atlas and loading option - run the sectioning processor and return the final Atlas.
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/README.md]: Nit: "raw atlas" -> "raw Atlas"

File: src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/README.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+2. Once the line slicing is completed - we need to look at any points that haven't been assigned a country code. These are Points that were not part of a Line feature - examples include stand alone Points such as trees, barriers, etc. This is one of the simpler slicing operations - since it's just a location containment check using the underlying grid index from the CountryBoundaryMap. 
+
+3. Once both Points and Lines have been sliced, at intermediate Atlas is built to be able to slice Relations. The intermediate changes are aggregated using a SimpleChangeSet and applied with the SimpleChangeSetHandler. It's important to note that during the build - we must replace any Lines that were sliced by their corresponding sliced Lines.
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/README.md]: Nit: "at intermediate Atlas" -> "an intermediate Atlas"

**Comment by adahn6** [comment]
Nice README files! Explanations were clear and thorough. Also a fan of the package change, the naming scheme makes more sense this way 💯

**Comment by MikeGost** [comment]
Going to re-open, might have been closed by mistake?

**Comment by adahn6** [comment]
wow I did not mean to close that. I did not even realize I had that power... sorry!!

File: src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/README.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+It's worth pointing out that some OSM Nodes will become both Atlas Points and Nodes. An example of this may be an OSM Node that has tagging and ends up at an intersection. In this case, it will become both an Atlas Point and an Atlas Node.
+
+### OSM Way to Atlas Line, Edge or Area
+
+For starters, we define an open OSM Way as a linear feature which does not share a first and last node. Conversely, a closed OSM Way has the last node shared with its first node. In general, the OSM Way to Atlas Entity translation is largely configuration driven. Please see the [pbf resource package](https://github.com/osmlab/atlas/tree/dev/src/main/resources/org/openstreetmap/atlas/geography/atlas/pbf) for a full view of default configurations. However, the generally accepted convention supported by the rest of the code base is that any navigable, open OSM Way (ex: highway, ferry, pier) becomes an Atlas Edge. An OSM Way that's non-navigable and open (ex: power line, wall) becomes an Atlas Line and anything that's non-navigable and closed becomes an Atlas Area.
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/README.md]: This should be a relative link here!

File: src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/README.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+## Terminology
+
+This section calls out the main terms and concepts that are key to the slicing code.
+
+* We rely on the JTS (Java Topology Suite) library for all geometric operations (spatial queries, intersection and cutting requests) and some key data structures (R-tree, JTS geometry representations).
+* The `CountryBoundaryMap` class is the main driver behind all slicing operations. This class contains the raw country boundary `MultiPolygon` representations as well as an underlying optimization (the grid index) that allows for quick lookup to see whether a specific feature is contained in a two dimensional rectangle.
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/README.md]: to see whether a specific feature is contained in a two dimensional rectangle. -> in a boundary

**Comment by matthieun** [comment]
Closing/ReOpening, looks like a Travis fluke.

**Comment by MikeGost** [comment]
Oops, I did the same as you!

**Pull request merged**